com.unity.render-pipelines.high-definition/Editor/Material/Nature/HDSpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,104 @@
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    /// <summary>
+    /// SpeedTree8 material upgrader for HDRP.
+    /// </summary>
+	class HDSpeedTree8MaterialUpgrader : SpeedTree8MaterialUpgrader
+    {
+        /// <summary>
+        /// Creates a SpeedTree8 material upgrader for HDRP.
+        /// </summary>
+        /// <param name="sourceShaderName">Original shader name.</param>
+        /// <param name="destShaderName">Upgrade shader name.</param>
+        public HDSpeedTree8MaterialUpgrader(string sourceShaderName, string destShaderName)
+			: base(sourceShaderName, destShaderName, HDSpeedTree8MaterialFinalizer)
+	{
+        }
+
+        public static void HDSpeedTree8MaterialFinalizer(Material mat)
+        {
+            SetHDSpeedTree8Defaults(mat);
+            SpeedTree8MaterialFinalizer(mat);
+            HDShaderUtils.ResetMaterialKeywords(mat);
+        }
+        /// <summary>
+        /// Checks if a given material is an HD SpeedTree8 material.
+        /// </summary>
+        /// <param name="mat">Material to check.</param>
+        /// <returns></returns>
+        public static bool IsHDSpeedTree8Material(Material mat)
+        {
+            return (mat.shader.name == "HDRP/Nature/SpeedTree8");
+        }
+
+        /// <summary>
+        /// Restores SpeedTree8-specific material properties and keywords that were set during import and should not be reset.
```

Comment:
Proposed rerwrite:
During import, the Editor may modify SpeedTree-specific material properties and keywords which should not be modified. This keyword restores these properties to their original state.

Reply:
What about:
"During normal usage, HDRP may reset SpeedTree-specific keywords which should not be modified. This method restores these properties to their original state."

Reply:
What is "normal usage"?

Reply:
When you click on a material for the first time after it's created, it resets that material's keywords. Comments say this is a hack and should probably not be mentioned specifically. 
Importing and upgrading a material also triggers a reset of that material keyword.

Reply:
"HDRP may reset SpeedTree-specific keywords which should not be modified. This method restores these keywords to their original state."

I propose leaving out the part about "normal usage" and just say it may reset things, because otherwise folks may want to dig into when and why.

Reply:
Sounds great, thanks!